Support tags

leahjlou opened the pull request:
- use custom title function in dialog, not just in pillbox
- highlight first item without having to press the down key when searching
- make search work on keyup
- add "Press enter to create X" text when `noRestrict` is true

Commit:
use custom title function in dropdown

Commit:
make sure first item is highlighted without having to press down key, make search work on keyup

dckesler commented on src/multi-selector.js:
```diff
@@ -79,32 +80,49 @@ const MultiSelector = React.createClass({
-				}, () => {
-					document.addEventListener("mousemove", this.state.mouseFunc);
-				})
+			if (!isNull(this.state.activeIndex) && !isNull(this.searchItems[this.state.activeIndex])) {
```
Would this.searchItems[this.state.activeIndex] be null? I might be wrong but it seems like it would be undefined

leahjlou replied:
good point, maybe I'll just check if it's truthy

dckesler commented on src/multi-selector.js:
```diff
@@ -79,32 +80,49 @@ const MultiSelector = React.createClass({
+	keyUp: function(e) {
+		const filterItems = this.getFilterItems(this.props.items);
+		const activeIndex = this.state.activeIndex;
+		this.props.onInputChange && this.props.onInputChange(e.currentTarget.value);
```
Is it okay that this only happen when the user takes his finger off the key?

leahjlou replied:
I haven't seen any usability weirdness but yeah technically the input changes on `keyDown` so I think i'll move this back

dckesler commented on src/multi-selector.js:
```diff
@@ -79,32 +80,49 @@ const MultiSelector = React.createClass({
+			if (isNull(activeIndex) && filterItems.length !== 0) {
 				return this.setActiveIndex(0);
-			} else {
+		 	} else {
```
I'm on my phone doing this, but is there a weird addition here before the else?

Commit:
allow adding new items when there's not an exact match, other cleanup

leahjlou commented:
@dckesler in the last commit I addressed a couple of your comments and also added something new, the comment should explain it. let me know if you have any questions

Commit:
change to onchange

dckesler commented:
🌵
